examples/notebooks/NYCTaxi-E2E-Demo/large_data.py

```diff
@@ -95,24 +95,35 @@ def clean(ddf, must_haves):
 start = dt()
 
 # apply a list of filter conditions to throw out records with missing or outlier values
+# perf hack; get rid of extra getitem calls
+fare_amount = taxi_df.fare_amount
+passenger_count = taxi_df.passenger_count
+pickup_longitude = taxi_df.pickup_longitude
+dropoff_longitude = taxi_df.dropoff_longitude
+pickup_latitude = taxi_df.pickup_latitude
+dropoff_latitude = taxi_df.dropoff_latitude
+trip_distance = taxi_df.trip_distance
+dropoff_datetime = taxi_df.dropoff_datetime
+pickup_datetime = taxi_df.pickup_datetime
+
 taxi_df = taxi_df[
-    (taxi_df.fare_amount > 1) &
-    (taxi_df.fare_amount < 500) &
-    (taxi_df.passenger_count > 0) &
-    (taxi_df.passenger_count < 6) &
-    (taxi_df.pickup_longitude > -75) &
-    (taxi_df.pickup_longitude < -73) &
-    (taxi_df.dropoff_longitude > -75) &
-    (taxi_df.dropoff_longitude < -73) &
-    (taxi_df.pickup_latitude > 40) &
-    (taxi_df.pickup_latitude < 42) &
-    (taxi_df.dropoff_latitude > 40) &
-    (taxi_df.dropoff_latitude < 42) &
-    (taxi_df.trip_distance > 0) &
-    (taxi_df.trip_distance < 500) &
-    ((taxi_df.trip_distance <= 50) | (taxi_df.fare_amount >= 50)) &

```

Comment:
BTW, pandas uses DataFrame._item_cache to speed up repeated getitem calls.  might be worth porting

Reply:
we should try I suppose